pkg/epp/datalayer/datasource.go

```diff
@@ -0,0 +1,211 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package datalayer
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+	"sync"
+)
+
+// DataSource is an interface required from all datalayer data collection
+// sources.
+type DataSource interface {
+	// Name returns the name of this datasource.
+	Name() string
+
+	// Start begins the collection process.
+	Start(ctx context.Context) error
+
+	// Stop stops the collection process.
+	Stop() error
+
+	// OutputType returns the type of information collected.
+	// TODO: this could be made private and excluded from the interface.
+	//       Use will likely be in the context of debugging/logging.
+	OutputType() reflect.Type
```

Comment:
can we keep it outside of the interface unless there is a real need to have it?
e.g., keep it now out, and if we see in next PRs it's needed put it back

Reply:
Yes - I will remove it.
The function to validate that data source output can be consumed by Extractor (based on its input type) can be called by the data source, passing it's output type.